components/homme/src/preqx_kokkos/prim_driver_mod.F90

```diff
@@ -98,32 +101,60 @@ end subroutine initialize_hommexx_session
     ! ==================================
     call prim_init1_elem_arrays(elem,par)
 
+    ! ==================================
+    ! Initialize the buffers for exchanges
+    ! ==================================
+    call prim_init1_buffers(elem,par)
```

Comment:
This will call 
```
    call initEdgeBuffer(par,edge_g,elem,max((qsize+3)*nlev+2,6*nlev+1))
```
which I think can be replaced with just nlev size buffer. Otherwise, this big edge_g buffer sits in the memory for the whole run (unless you freed it and I did not notice), but it is only needed for 1 test init. Maybe, freeing it after tests init is also an option? unless it introduces even more infrastructure diffs.

Reply:
Good call. If the init only needs a small buffer, I'll go ahead and change this to an ad-hoc call to initEdgeBuffer. Free-ing it after initialization is also an option, but it may be more code edits than simply allocate an nlev-wide buffer.